sisu/src/main/java/io/smallrye/beanbag/sisu/Sisu.java

```diff
@@ -120,7 +120,8 @@ private void addPlexusComponents(ClassLoader classLoader, DependencyFilter filte
         try (InputStream is = conn.getInputStream()) {
             try (InputStreamReader isr = new InputStreamReader(is, StandardCharsets.UTF_8)) {
                 try (BufferedReader br = new BufferedReader(isr)) {
-                    XMLStreamReader xr = XMLInputFactory.newInstance().createXMLStreamReader(br);
+                    XMLStreamReader xr = XMLInputFactory.newFactory(XMLInputFactory.class.getName(), classLoader)
+                            .createXMLStreamReader(br);
```

Comment:
> Actually, I'm not reproducing my original issue anymore but with my safeguards in place, I can reproduce the fact that SmallRye BeanBag tries to use a closed CL to load some resources/classes.

If it's all about using a closed CL due to service loader shenanigans, then I'd recommend this simpler solution:

```suggestion
                    XMLStreamReader xr = XMLInputFactory.newDefaultFactory()
                            .createXMLStreamReader(br);
```

This avoids class loading completely, and just directly instantiates the factory from the JDK.

Reply:
@dmlloyd I adjusted the PR to follow your suggestion and the tests pass and the issue is solved in Quarkus too so I would say let's go for it.

@aloubyansky I let you have a look too.

If things are fine for both of you, I would appreciate a quick release so I can iterate on my side. Thanks!